### PR TITLE
community[minor]: Add Moonshot chat models integration

### PR DESCRIPTION
**Pull request by crazyurus**
[Moonshot](https://platform.moonshot.cn/) is a Chinese startup that provides LLM service for companies and individuals. See https://platform.moonshot.cn/docs/intro.

Langchain Python already has Moonshot integration. See https://python.langchain.com/docs/integrations/chat/moonshot/

So complete this integration in the Javascript version

### CODE REVIEW

File: libs/langchain-community/src/chat_models/moonshot.ts [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+  n?: number;
+
+  constructor(fields: Partial<ChatMoonshotParams> & BaseChatModelParams = {}) {
+    super(fields);
+
+    this.moonshotApiKey =
+      fields?.apiKey ??
+      fields?.moonshotApiKey ??
+      getEnvironmentVariable("Moonshot_API_KEY");
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/moonshot.ts:239]: Should be all caps

File: libs/langchain-community/src/chat_models/moonshot.ts [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+    return "ChatMoonshot";
+  }
+
+  get callKeys() {
+    return ["stop", "signal", "options"];
+  }
+
+  get lc_secrets() {
+    return {
+      moonshotApiKey: "Moonshot_API_KEY",
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/moonshot.ts:196]: Change to `MOONSHOT_API_KEY`

File: libs/langchain-community/src/chat_models/moonshot.ts [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+    };
+  }
+
+  get lc_aliases() {
+    return undefined;
+  }
+
+  moonshotApiKey?: string;
+
+  apiKey?: string;
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/moonshot.ts:207]: Why two different variables?

File: libs/langchain-community/src/chat_models/moonshot.ts (hunk 1 above)
Comment (crazyurus) [reply libs/langchain-community/src/chat_models/moonshot.ts]: Yes, environment variables need to be capitalized. Already updated.

File: libs/langchain-community/src/chat_models/moonshot.ts (hunk 2 above)
Comment (crazyurus) [reply libs/langchain-community/src/chat_models/moonshot.ts]: Same as above

File: libs/langchain-community/src/chat_models/moonshot.ts [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+          }
+        }
+      }
+    };
+
+    return this.caller.call(makeCompletionRequest);
+  }
+
+  _llmType(): string {
+    return "Moonshot";
```
Comment (jacoblee93) [inline libs/langchain-community/src/chat_models/moonshot.ts:463]: nit: lower-case

**Comment by jacoblee93** [comment]
Hey, I'm very sorry for the delay - I was on vacation and then we had a lot to do for the launch of 0.2.0. Small comments but looks good otherwise! I can update this tomorrow unless you beat me to it.

File: libs/langchain-community/src/chat_models/moonshot.ts (hunk 3 above)
Comment (crazyurus) [reply libs/langchain-community/src/chat_models/moonshot.ts:195]: `moonshotApiKey` removed.

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**